File: algorithms/levenstein-distance/levenshtein.py
#!/usr/bin/python

def levenshtein(w1, w2):
	w1 = w1.lower()
	w2 = w2.lower()
	n1 = len(w1) + 1
	n2 = len(w2) + 1
	d = [[0 for x in range(n2)] for y in range(n1)]
	
	for y in range(0, n1):
		d[y][0] = y
	for x in range(0, n2):
		d[0][x] = x

	for y in range(1, n1):
		for x in range(1, n2):
			if w1[y-1] == w2[x-1]:
				d[y][x] = d[y-1][x-1]
			else:
				d[y][x] = min(d[y-1][x-1], d[y-1][x], d[y][x-1]) + 1
	
	return d[n1-1][n2-1]

print(levenshtein("Roman", "Ramon"))

File: algorithms/levenshtein-distance/levenshtein.php
<?php
	declare(strict_types = 1);
	
	/*
		there is a buildin method `levenshtein`, see http://docs.php.net/manual/en/function.levenshtein.php
		code here is for the lulz
	*/

	function levenshtein1(string $w1, string $w2) : int {
		$w1 = strtolower($w1);
		$w2 = strtolower($w2);
		$n1 = strlen($w1) + 1;
		$n2 = strlen($w2) + 1;
		$d = array_fill(0, $n1, array_fill(0, $n2, 0));

		for($y=1; $y<$n1; $y++) $d[y][0] = y;
		for($x=1; $x<$n2; $x++) $d[0][x] = x;

		for($y=1; $y<$n1; $y++) {
			for($x=1; $x<$n2; $x++) {
				if($w1[$y-1] ==  $w2[$x-1]) {
					$d[$y][$x] = $d[$y-1][$x-1];
				} else {
					$d[$y][$x] = min($d[$y-1][$x-1], $d[$y-1][$x], $d[$y][$x-1]) + 1;
				}
			}
		}
	
		return $d[$n1-1][$n2-1];
	}

	echo levenshtein1("Roman", "Ramon");
?>
